nextest-runner/src/platform.rs

```diff
@@ -5,64 +5,374 @@
 
 use crate::{
     cargo_config::{CargoTargetArg, TargetTriple},
-    errors::{TargetTripleError, UnknownHostPlatform},
+    errors::{RustBuildMetaParseError, TargetTripleError, UnknownHostPlatform},
 };
+use camino::Utf8PathBuf;
+use nextest_metadata::BuildPlatformSummary;
+use std::io::{self, Cursor};
 use target_spec::summaries::PlatformSummary;
 pub use target_spec::Platform;
 
-/// A representation of host and target platforms.
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub struct BuildPlatforms {
-    /// The host platform.
-    pub host: Platform,
+fn read_first_line_as_path(mut reader: impl io::BufRead) -> Option<Utf8PathBuf> {
+    // We will print warn logs later when we are adding the path to the dynamic linker search paths,
+    // so we don't print the warn log here to avoid spammy log.
+    let mut line = Default::default();
+    match reader.read_line(&mut line) {
+        Ok(_) => {
+            let original_line = line.as_str();
+            let line = line.trim();
+            if line.is_empty() {
+                log::debug!("empty input found: {:#?}", original_line);
+                return None;
+            }
+            Some(Utf8PathBuf::from(line))
+        }
+        Err(e) => {
+            log::debug!("failed to read the input: {:#?}", e);
+            None
+        }
+    }
+}
 
-    /// The target platform, if specified.
-    ///
-    /// In the future, this will become a list of target triples once multiple `--target` arguments
-    /// are supported.
-    pub target: Option<TargetTriple>,
+/// Creates a new [`BuildPlatform`].
+#[derive(Default)]
+pub struct BuildPlatformBuilder {
+    host_libdir: Option<Utf8PathBuf>,
+    target: Option<BuildPlatformTarget>,
 }
 
-impl BuildPlatforms {
-    /// Creates a new `BuildPlatform`.
+impl BuildPlatformBuilder {
+    /// Creates a new [`BuildPlatformTargetBuilder`] used to populate a [`BuildPlatformTarget`] in
+    /// the final built [`BuildPlatform`].
+    ///
+    /// Returns an error if the host platform could not be determined.
+    pub fn set_target(&mut self, triple: TargetTriple) -> BuildPlatformTargetBuilder<'_> {
+        BuildPlatformTargetBuilder {
+            libdir: None,
+            parent: self,
+            triple,
+        }
+    }
+
+    /// Set the host libdir to the built [`BuildPlatform`].
     ///
+    /// Used to set the dynamic linker search path when running the test executables. If we fail to
+    /// parse the input, the libdir won't be set.
+    pub fn set_host_libdir(&mut self, reader: impl io::BufRead) {
```

Comment:
I'd rather this just take a `Utf8PathBuf`, seems a bit too implementation-dependent to take the stdout of rustc here.

Reply:
Changed to `set_host_libdir_from_rustc_output` for setting from the output. If we just want to set it, just use mutate the struct. WDYT?